pages/api/auth/security-questions/index.ts

```diff
@@ -0,0 +1,20 @@
+import { middleware, cors } from "@lib/middleware";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { logMessage } from "@lib/logger";
+import { retrieveUserSecurityQuestions } from "@lib/auth";
+
+const apiHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const email = req.query.email;
+
+  if (!email || Array.isArray(email)) return res.status(400).json({ error: "Malformed request" });
+
+  try {
+    const securityQuestions = await retrieveUserSecurityQuestions({ email });
```

Comment:
This unprotected route reveals exactly which questions the user has provided answers to.  Could this info be passed through getServerSide props instead?

Reply:
@timarney if you tell me this is possible then I could also remove that other endpoint which would not be used anymore.

@bryan-robitaille I originally thought this would work as I found other websites allowing you to see/answer questions as long as you know the user email/username.

Reply:
It's a pretty big security hole and would invalidate our security controls.

Reply:
As discussed we need to add a email verify step prior to displaying that page.

Making adjustments so we'll be able to remove that endpoint. 

https://github.com/cds-snc/platform-forms-client/pull/2496